classes/announcement/PKPAnnouncementDAO.inc.php

```diff
@@ -339,6 +339,7 @@ function &getAnnouncementsNotExpiredByAssocId($assocType, $assocId, $rangeInfo =
 			WHERE assoc_type = ?
 				AND assoc_id = ?
 				AND (date_expire IS NULL OR date_expire > CURRENT_DATE)
+				AND (date_posted < CURRENT_DATE)
```

Comment:
The bug reporter's goal was to permit setting a future date and having the announcement not get published until then.